analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.cs

```diff
@@ -29,5 +29,81 @@ public class PublicMethodArgumentsShouldBeCheckedForNull : SymbolicRuleCheck
 
     protected override DiagnosticDescriptor Rule => S3900;
 
-    public override bool ShouldExecute() => false;
+    public override bool ShouldExecute()
+    {
+        return IsAccessibleFromOtherAssemblies(Node)
+            && (IsRelevantMethod(Node) || IsRelevantPropertyAccessor(Node));
+
+        static bool IsRelevantMethod(SyntaxNode node) =>
+            node is BaseMethodDeclarationSyntax { ParameterList.Parameters.Count: > 0 } method
+            && MethodDereferencesArguments(method);
+
+        static bool IsRelevantPropertyAccessor(SyntaxNode node) =>
+            node is AccessorDeclarationSyntax accessor
+            && IsPropertyAccessorAccessibleFromOtherAssemblies(accessor.Modifiers);
+
+        static bool IsAccessibleFromOtherAssemblies(SyntaxNode node) =>
+            node.AncestorsAndSelf().OfType<MemberDeclarationSyntax>().FirstOrDefault() is { } containingMember
+            && node.Ancestors().OfType<BaseTypeDeclarationSyntax>().FirstOrDefault() is { } containingType
+            && IsMemberAccessibleFromOtherAssemblies(containingMember.Modifiers(), containingType)
+            && IsTypeAccessibleFromOtherAssemblies(containingType.Modifiers);
+
+        static bool IsMemberAccessibleFromOtherAssemblies(SyntaxTokenList modifiers, BaseTypeDeclarationSyntax containingType) =>
+            IsPublicOrProtectedOrProtectedInternal(modifiers)
+            || (containingType is InterfaceDeclarationSyntax && HasNoDeclaredAccessabilityModifier(modifiers));
+
+        static bool IsPropertyAccessorAccessibleFromOtherAssemblies(SyntaxTokenList modifiers) =>
+            IsPublicOrProtectedOrProtectedInternal(modifiers)
+            || HasNoDeclaredAccessabilityModifier(modifiers);
+
+        static bool IsTypeAccessibleFromOtherAssemblies(SyntaxTokenList modifiers) =>
+            IsPublicOrProtectedOrProtectedInternal(modifiers);
+
+        static bool IsPublicOrProtectedOrProtectedInternal(SyntaxTokenList modifiers) =>
+            modifiers.AnyOfKind(SyntaxKind.PublicKeyword)
+            || (modifiers.AnyOfKind(SyntaxKind.ProtectedKeyword) && !modifiers.AnyOfKind(SyntaxKind.PrivateKeyword));
+
+        static bool HasNoDeclaredAccessabilityModifier(SyntaxTokenList modifiers) =>
```

Comment:
```suggestion
        static bool HasNoDeclaredAccessibilityModifier(SyntaxTokenList modifiers) =>
```

Reply:
![image](https://user-images.githubusercontent.com/103252490/226590297-9d2ea72c-32ee-45bb-a5b1-38bd233fb50e.png)

Reply:
Oops 🤣 I mistook switched the lines in the Commit suggestion and thought yours was the wrong one. Mea culpa! Corrected it.